spec/system/nested_documentable_spec.rb

```diff
@@ -0,0 +1,314 @@
+      visit path
+
+      documentable_attach_new_file(file_fixture("empty.pdf"))
+      click_link "Remove document"
+
+      expect(page).not_to have_css("#nested-documents .document-fields")
+    end
+
+    scenario "Should show successful notice when resource filled correctly without any nested documents" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      fill_in_required_fields(factory, path)
+      click_button submit_button_text
+
+      expect(page).to have_content notice_text
+    end
+
+    scenario "Should show successful notice when resource filled correctly and after valid file uploads" do
+      Setting["uploads.documents.max_amount"] = 3
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      fill_in_required_fields(factory, path)
+
+      %w[clippy empty logo].take(documentable.class.max_documents_allowed).each do |filename|
+        click_link "Add new document"
+
+        attach_file "Choose document", file_fixture("#{filename}.pdf")
+
+        within all(".document-fields").last do
+          expect(page).to have_css ".loading-bar.complete"
+        end
+      end
+      click_button submit_button_text
+
+      if factory != :dashboard_action
+        within "#documents" do
+          expect(page).to have_content "Documents (3)"
+          expect(page).to have_link text: "empty.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "empty.pdf"
+          expect(page).to have_link text: "clippy.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "clippy.pdf"
+          expect(page).to have_link text: "logo.pdf"
+          expect(page).to have_css "a[rel=nofollow]", text: "logo.pdf"
+          expect(page).not_to have_css "a[target=_blank]"
+        end
+      end
+      expect(page).to have_content notice_text
+    end
+
+    describe "Metadata" do
+      let(:factory) { (factories - [:dashboard_action]).sample }
+
+      scenario "download document without metadata" do
+        do_login_for(user, management: management_section?(path))
+        visit path
+
+        fill_in_required_fields(factory, path)
+        documentable_attach_new_file(file_fixture("logo_with_metadata.pdf"))
+        click_button submit_button_text
+
+        expect(page).to have_content notice_text
+
+        io = URI.parse(find_link(text: "PDF")[:href]).open
+        reader = PDF::Reader.new(io)
+
+        expect(reader.info[:Keywords]).not_to eq "Test Metadata"
+        expect(reader.info[:Author]).not_to eq "Test Developer"
+        expect(reader.info[:Title]).not_to eq "logo_with_metadata.pdf"
+        expect(reader.info[:Producer]).not_to eq "Test Producer"
+        expect(reader.info).to eq({})
+      end
+    end
+
+    describe "When allow attached documents setting is disabled" do
+      before do
+        Setting["feature.allow_attached_documents"] = false
+        create(:document, documentable: documentable)
```

Comment:
Note that, with this line, the "Add new document" link wouldn't be present even if the setting was enabled :thinking:. Maybe we should keep separate tests for the  "Add new document" link and the "#documents" element :thinking:. What do you think?